pkg/api/types.go

```diff
@@ -1407,6 +1409,69 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimit is a rate limit on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service
+	RateLimit *RateLimitSpec `json:"ratelimit,omitempty"`
```

Comment:
@mml 
> 
2. We should reject budgets with >1 rate limit at the API server.
3. We should make it clear that other clients should properly deal with any number of rate limits rather than expecting to always see 1 rate limit.

I am a little confused.  I think API server could handle the multi-ratelimit situation gracefully: the "evict" subresource will ensure that a disruption could only happen when it meet ALL the ratelimits.
So, why should we reject budgets with >1 rate limit at the API server?